pages/[username].tsx

```diff
@@ -0,0 +1,198 @@
+import LayoutContainer from "@components/LayoutContainer";
+import RedirectUtils from "utils/redirect";
+import Toggle from 'react-toggle'
+import UserAgentUtils from "utils/userAgent";
+import getConfig from "next/config";
+import parser from 'ua-parser-js'
+import useToastHook from "@hooks/toast/useToastHook"
+import React, { useEffect, useRef, useState } from "react";
+import { BaseContainer } from "@components/Page/BaseContainer";
+import { GetServerSidePropsContext } from "next";
+import { Header } from "@components/Header";
+import { Helmet } from "react-helmet";
+import { LoaderWrapper } from "@components/LoaderWrapper";
+import { MessageEnum } from "@shared/enum";
+import { toast } from "react-toastify";
+import { useGetProfile } from "@services/profile/profileHooks";
+import { useRouter } from "next/router";
+
+interface PageProps {
+    username?: string
+    isDynamicLink?: boolean
+}
+
+const { publicRuntimeConfig } = getConfig()
+
+export default function Profile(props: PageProps) {
+    const {isDynamicLink, username} = props
+    const { data, isLoading } = useGetProfile(username);
+    const [message, setMessage] = useState('')
+    const router = useRouter();
+    useEffect(() => {
+        if (data) {
+            localStorage.setItem(MessageEnum.targetUser, data.user_id);
+        }
+    }, [data])
+
+    useEffect(() => {
+        if (isDynamicLink) router.push(`/${username}`, undefined, { shallow: true })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [isDynamicLink])
+
+    const textAreaRef = useRef<HTMLTextAreaElement>(null)
+
+    useEffect(() => {
+        adjustTextareaHeight();
+    }, [message]);
+
+    const adjustTextareaHeight = () => {
+        if (textAreaRef.current) {
+            const lineHeight = parseFloat(getComputedStyle(textAreaRef.current).lineHeight);
+            textAreaRef.current.style.height = `${lineHeight}px`;
+            const lines = Math.floor(textAreaRef.current.scrollHeight / lineHeight);
+            const numLines = lines <= 3 ? lines : 3;
+            textAreaRef.current.style.height = `${numLines * lineHeight}px`;
+        }
+    };
+
+    const { copyToClipboardToast } = useToastHook()
+
+    const showDownloadToast = () => toast('Download the Helio app now to create a profile and send incognito messages', {
+        position: "bottom-center",
+        autoClose: 3000,
+        hideProgressBar: true,
+        closeOnClick: true,
+        pauseOnHover: true,
+        theme: "dark",
+        closeButton: false,
+        style: {
+            borderRadius: '8px',
+            width: '270px',
+            margin: 'auto',
+            marginBottom: '60px',
+            textAlign: 'center',
+            backgroundColor: 'rgba(0, 0, 0, 0.25)'
+        },
+    })
+
+    const onShareButtonClicked = () => {
+        navigator.clipboard.writeText(`${publicRuntimeConfig.DYNAMIC_LINK_DOMAIN}/${username}`)
+        copyToClipboardToast()
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This code assumes that the `navigator.clipboard.writeText` function will always succeed. It would be better to handle potential errors with a try-catch block.

```diff
    const onShareButtonClicked = () => {
        try {
            navigator.clipboard.writeText(`${publicRuntimeConfig.DYNAMIC_LINK_DOMAIN}/${username}`)
            copyToClipboardToast()
        } catch (error) {
            console.error('Failed to copy text: ', error);
        }
    }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->